### PR TITLE
Fix canonical URL and og:url.

### LINKED ISSUE #1180: Fix canonical URL and og:url

### PR DESCRIPTION
**Pull request by XhmikosR**
Fixes #1180

### CODE REVIEW

File: routes/appendLocals.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ function generateSRI(file) {
 
 function appendLocals(req, res) {
     const siteUrl = getCurrentSiteurl(req);
-    const canonicalUrl = `${config.siteurl}${req.path}`;
+    const canonicalUrl = `${config.siteurl}${req.originalUrl}`;
```
Comment (XhmikosR) [inline routes/appendLocals.js]: I'm pretty sure this worked up some point.

**Comment by XhmikosR** [comment]
I'll try to add tests for these tomorrow.

**Comment by XhmikosR** [comment]
One thing left: should we distinguish pages based on the `theme` param? Because that's what we are doing.

File: routes/appendLocals.js (hunk 1 above)
Comment (XhmikosR) [reply routes/appendLocals.js]: OK, this broke after b0238932d9c0b9fa5c904cf990bdf29e7a814922

Inside say about.js:

```js
console.log(req.originalUrl);   // /about/
console.log(req.path);          // /
console.log(req.pathname);      // undefined
console.log(req.url);           // /
```

I'm not sure how to fix this and keep this modular structure...

File: views/_partials/head/social.pug [hunk 2]
```diff
@@ -1,7 +1,7 @@
 meta(property='og:title', content!=title)
 meta(property='og:type', content='website')
 meta(property='og:description', content!=pageDescription)
-meta(property='og:url', content=siteUrl)
+meta(property='og:url', content=siteUrl + pageUrl)
```
Comment (jmervine) [inline views/_partials/head/social.pug]: Shouldn't this be 
```
content=`${siteUrl}${pageUrl}`
```

File: routes/appendLocals.js (hunk 1 above)
Comment (jmervine) [reply routes/appendLocals.js]: Is it working as expected with this change to `req.originalUrl`?

Comment (XhmikosR) [reply routes/appendLocals.js]: Yes, but `originalUrl` contains the query params which isn't what we want. I verified that this happens after  the aforementioned commit. For example, [this](https://github.com/MaxCDN/bootstrapcdn/tree/develop-xmr-before-modularize-app) branch doesn't have the canonical URL issue.

So, something is overwritten somewhere or something like that.

Comment (XhmikosR) [reply routes/appendLocals.js]: If we don't find a solution soon, I'll need to revert the whole modularization and I don't want to do it :/

But I don't know how to solve this issue. I could possibly work around it by using `url.parse`, but I feel we are doing something wrong in the Router.

**Comment by XhmikosR** [comment]
@jmervine: I added a workaround. I know I should add tests too but I don't have a lot of time. It should fix the issues though.

**Pull request merged**